Add cloud resources for Datadog integration (#1604)
port-labs/port-docs: docs/guides/all/ingest-cloud-resources-using-datadog.md

---
displayed_sidebar: null
description: Learn how to ingest cloud resources using Datadog in Port, enhancing visibility and performance monitoring.
---

# Ingest cloud resources with Datadog

This guide aims to show you how to ingest cloud resources using Datadog to have a good grasp of the cloud resources/entities you have from your cloud provider.

## Common use cases
- Map your monitored resources from cloud providers in Datadog

## Prerequisites
This guide assumes the following:
- You have a Port account and that you have finished the [onboarding process](/quickstart).
- You have [installed and setup Port's Datadog integration](/build-your-software-catalog/sync-data-to-catalog/apm-alerting/datadog.md)
- You have installed integrations for [GCP](https://docs.datadoghq.com/integrations/google_cloud_platform/?tab=project), [AWS](https://docs.datadoghq.com/integrations/#cat-aws) and [Azure](https://docs.datadoghq.com/integrations/azure/) on your Datadog environment

## Ingesting cloud resources into Port

> We will be making use of the `hosts` kind in Port's Datadog integration which provides information on hosts existing on the cloud providers we have configured. So far, we are only interested in hosts with `gcp`, `aws`, and `azure` in the `sources` property.

Reviewer: ```suggestion
We will be making use of the `hosts` kind in Port's Datadog integration which provides information on hosts existing on the cloud providers we have configured. For this example, we are only interested in hosts with `gcp`, `aws`, and `azure` in the `sources` property.
```

1. Having installed the Datadog integration, create the `datadogCloudResource` blueprint in your Port environment using the blueprint below:

<details>

> <summary><b>Datadog Cloud Resource Blueprint (Click to expand</b></summary>

Reviewer 2: ```suggestion
<summary><b>Datadog Cloud Resource Blueprint (Click to expand)</b></summary>
```

```yaml showLineNumbers

deleteDependentEntities: true
createMissingRelatedEntities: true
resources:
  - kind: host
    selector:
      query: >-
        [.sources[] | . as $source | ["azure", "gcp", "gce", "aws"] |
        contains([$source])] | any(.)
    port:
      entity:
        mappings:
          blueprint: '"datadogCloudResource"'
          identifier: .id | tostring
          title: .aws_name // .host_name
          properties:
            up: .up
            host_name: .host_name
            platform: .meta.platform
            is_muted: .is_muted
            machine: .meta.machine
            description: .description
            sources: .sources
            cpu_cores: .meta.cpuCores
            agent_version: .meta.agent_version
            tags: .tags_by_source

```

</details>

:::info Mapping explanation
Cloud resources from Datadog are hosts from the `host` kind of the Datadog integration which are from GCP, AWS or Azure. To ingest them, we have to specify a selector which allows only such hosts with the sources we specify.

By doing this, we are able to ingest all cloud resources from Datadog

:::

> 4. Next, click on resync and watch your cloud resources from Datadog being ingested:

Reviewer: ```suggestion
4. Finally, click `resync`. Your cloud resources from Datadog should now be ingested:
```
